File: front-end/src/app/models/DepartmanNastavnik.ts
import { Departman } from "./Departaman";
import { Nastavnik } from "./Nastavnik";

export interface DepartmanNastavnik {
  id?: number | null;
  vidljiv: boolean;
  departman: Departman;
  nastavnik: Nastavnik;
  
}

File: back-end/src/main/java/server/model/DepartmanNastavnik.java
package server.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class DepartmanNastavnik {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(optional = false)
    private Departman departman;

    @ManyToOne(optional = false)
    private Nastavnik nastavnik;

    @Column(nullable = false)
    private Boolean vidljiv = true;
}